### DISCUSSION TITLE
hue dimmer behaviour change after changing "legacy" option

### DISCUSSION THREAD

**Discussion by hugalafutro** [Koenkk/zigbee2mqtt#17574] [Devices]
Long story short I had some users of my Home Assistant Dimmer blueprint had problems with it and we narrowed it down to payloads using in some cases underscores and in others dashes. This was due to me having the "legacy" option on while creating the blueprint, while others had it off. So I turned the legacy option off, and adjusted my blueprint and all is sort of well.

I use the on off buttons as following:
on click - turn on ceiling lamp
on hold - turn on all lamps
off click -  turn off ceiling lamp
off hold - turn off all lamps

After switching the "legacy" option off in z2m settings for the Dimmer, what now happens is:
on click - works as previously
on hold - immediately performs the "on click" action when pressing, then triggers the "on hold" action half a second later
off click - works as previously
off hold - you get the idea...

While, yes, when it comes to lights this is just a minor visual annoyance as to the order in which the lights come on/off, I have other button that controls media player on click and window blind on hold, so when I want to raise the blind it first pauses my media player and that is no bueno.

How can I make it not trigger the "on click" when triggering the "on hold" ?

Also, I am somewhat confused by the description, if it's "highly recommended" to be false, why is default true ?
```
legacy
Set to false to disable the legacy integration (highly recommended), will change structure of the published payload (default true).
```

Thanks for any insights!

**Comment by niklaspeura** [comment]
Did you ever figure out what the legacy integration does?
I have a similar issue where one of my Styrbar remotes doesn't expose the brightness_move_up (down/stop) actions anymore.

**Reply by hugalafutro** [reply]
Nope, I just resigned to live with it. I don't use it that often, but it annoys the hell out of me every single time I do.
